### PR TITLE
fix: add various fixes to parser and generator

### PR DESCRIPTION
**Pull request by Lodin**

### CODE REVIEW

File: packages/ts/generator-typescript-plugin-backbone/src/EndpointProcessor.ts [hunk 1]
```diff
@@ -6,16 +7,15 @@ import { OpenAPIV3 } from 'openapi-types';
 import type { ReadonlyDeep } from 'type-fest';
 import type { SourceFile, Statement } from 'typescript';
 import EndpointMethodOperationProcessor, { EndpointMethodOperation } from './EndpointMethodOperationProcessor.js';
-import type { BackbonePluginContext } from './utils.js';
 
 export default class EndpointProcessor {
-  readonly #context: BackbonePluginContext;
+  readonly #context: Plugin;
```
Comment (haijian-vaadin) [inline packages/ts/generator-typescript-plugin-backbone/src/EndpointProcessor.ts:12]: should the `context` be renamed to `plugin`?

File: packages/ts/generator-typescript-plugin-backbone/src/EndpointMethodResponseProcessor.ts [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-import type { BackbonePluginContext } from './utils.js';
 import { defaultMediaType } from './utils.js';
 
 export type EndpointMethodResponses = ReadonlyDeep<OpenAPIV3.ResponsesObject>;
 export type EndpointMethodResponse = ReadonlyDeep<OpenAPIV3.ResponseObject>;
 
 export default class EndpointMethodResponseProcessor {
   readonly #code: string;
-  readonly #context: BackbonePluginContext;
+  readonly #context: Plugin;
```
Comment (haijian-vaadin) [inline packages/ts/generator-typescript-plugin-backbone/src/EndpointMethodResponseProcessor.ts:14]: should the `context` be renamed to `plugin`?

File: packages/ts/generator-typescript-plugin-backbone/src/EndpointMethodRequestBodyProcessor.ts [hunk 3]
```diff
@@ -21,14 +21,14 @@ export type EndpointMethodRequestBodyProcessingResult = Readonly<{
 }>;
 
 export default class EndpointMethodRequestBodyProcessor {
-  readonly #context: BackbonePluginContext;
+  readonly #context: Plugin;
```
Comment (haijian-vaadin) [inline packages/ts/generator-typescript-plugin-backbone/src/EndpointMethodRequestBodyProcessor.ts:24]: should the `context` be renamed to `plugin`?

File: packages/ts/generator-typescript-plugin-backbone/src/EndpointMethodOperationProcessor.ts [hunk 4]
```diff
@@ -26,7 +26,7 @@ export default abstract class EndpointMethodOperationProcessor {
     endpointMethodName: string,
     operation: EndpointMethodOperation,
     dependencies: DependencyManager,
-    context: BackbonePluginContext,
+    context: Plugin,
```
Comment (haijian-vaadin) [inline packages/ts/generator-typescript-plugin-backbone/src/EndpointMethodOperationProcessor.ts:29]: should the `context` be renamed to `plugin`?

File: packages/ts/generator-typescript-plugin-backbone/src/EndpointMethodResponseProcessor.ts [hunk 5]
```diff
@@ -1,25 +1,25 @@
...
   readonly #dependencies: DependencyManager;
   readonly #response: EndpointMethodResponse;
 
   public constructor(
     code: string,
     response: EndpointMethodResponses[string],
     dependencyManager: DependencyManager,
-    context: BackbonePluginContext,
+    context: Plugin,
```
Comment (haijian-vaadin) [inline packages/ts/generator-typescript-plugin-backbone/src/EndpointMethodResponseProcessor.ts:22]: should the `context` be renamed to `plugin`?

File: packages/ts/generator-typescript-plugin-backbone/src/EntityProcessor.ts [hunk 6]
```diff
@@ -22,23 +23,17 @@ import { dirname } from 'path/posix';
...
-
-const exportDefaultModifiers = [
-  ts.factory.createModifier(ts.SyntaxKind.ExportKeyword),
-  ts.factory.createModifier(ts.SyntaxKind.DefaultKeyword),
-];
 
 export class EntityProcessor {
   readonly #component: Schema;
-  readonly #context: BackbonePluginContext;
+  readonly #context: Plugin;
```
Comment (haijian-vaadin) [inline packages/ts/generator-typescript-plugin-backbone/src/EntityProcessor.ts:29]: should the `context` be renamed to `plugin`?

File: packages/ts/generator-typescript-plugin-backbone/src/EntityProcessor.ts [hunk 7]
```diff
@@ -22,23 +23,17 @@ import { dirname } from 'path/posix';
...
   readonly #dependencies;
   readonly #fullyQualifiedName: string;
   readonly #name: string;
   readonly #path: string;
   readonly #sourcePaths = new PathManager({ extension: 'ts' });
 
-  public constructor(name: string, component: Schema, context: BackbonePluginContext) {
+  public constructor(name: string, component: Schema, context: Plugin) {
```
Comment (haijian-vaadin) [inline packages/ts/generator-typescript-plugin-backbone/src/EntityProcessor.ts:36]: should the `context` be renamed to `plugin`?

File: packages/ts/generator-typescript-plugin-barrel/src/BarrelProcessor.ts [hunk 8]
```diff
@@ -1,24 +1,24 @@
...
 import { basename, dirname } from 'path';
-import type Pino from 'pino';
 import type { SourceFile } from 'typescript';
 
 export default class BarrelProcessor {
   public static readonly BARREL_FILE_NAME = 'endpoints.ts';
   public declare ['constructor']: typeof BarrelProcessor;
   readonly #endpoints: readonly SourceFile[];
-  readonly #logger: Pino.Logger;
+  readonly #context: Plugin;
```
Comment (haijian-vaadin) [inline packages/ts/generator-typescript-plugin-barrel/src/BarrelProcessor.ts:12]: should the `context` be renamed to `plugin`?

File: packages/ts/generator-typescript-plugin-barrel/src/BarrelProcessor.ts [hunk 9]
```diff
@@ -1,24 +1,24 @@
...
   readonly #sourcePaths = new PathManager({ extension: 'ts' });
 
-  public constructor(endpoints: readonly SourceFile[], logger: Pino.Logger) {
+  public constructor(endpoints: readonly SourceFile[], context: Plugin) {
```
Comment (haijian-vaadin) [inline packages/ts/generator-typescript-plugin-barrel/src/BarrelProcessor.ts:15]: should the `context` be renamed to `plugin`?

File: packages/ts/generator-typescript-plugin-client/src/ClientProcessor.ts [hunk 10]
```diff
@@ -1,22 +1,22 @@
...
 import DependencyManager from '@vaadin/generator-typescript-utils/dependencies/DependencyManager.js';
 import PathManager from '@vaadin/generator-typescript-utils/dependencies/PathManager.js';
-import type Pino from 'pino';
 import type { SourceFile } from 'typescript';
 import ts from 'typescript';
 
 export default class ClientProcessor {
   readonly #filePath: string;
-  readonly #logger: Pino.Logger;
+  readonly #context: Plugin;
```
Comment (haijian-vaadin) [inline packages/ts/generator-typescript-plugin-client/src/ClientProcessor.ts:11]: should the `context` be renamed to `plugin`?

**Review by haijian-vaadin — CHANGES_REQUESTED** [review]

File: packages/ts/generator-typescript-plugin-backbone/src/EndpointProcessor.ts (hunk 1 above)
Comment (Lodin) [reply packages/ts/generator-typescript-plugin-backbone/src/EndpointProcessor.ts:12]: What about `owner`?

**Review by haijian-vaadin — DISMISSED** [review]

**Review by haijian-vaadin — DISMISSED** [review]

**Review by haijian-vaadin — APPROVED** [review]

**Pull request merged**